### PR TITLE
Fix: #13904

### LINKED ISSUE #13904: README.md: Commits signing not mentioned
## What page did you find the problem on?
https://github.com/mdn/content/blob/main/README.md

## What is the problem?
No mention of [Signing the commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) while contributing using GitHub UI or git on local machine. New contributors may find out this requirement the hard way .

## What did you expect to see?
- mention of: all the commits need to be signed in order to be accepted.
- mention of: while contributing using GitHub UI, [GPG key](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account) need to be provided in settings.
- mention of: while using git tool on local machine [sign commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- relevant links to setup the signatures like: https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account 
- in the README.md update all the `commit` statements to have `-S` flag to sign the commits.
```
commit -S -m "fix: message"
```

### PR DESCRIPTION
**Pull request by MrBrain295**
#### Summary

Mention that you need a GPG key signed commit to have your PR accsepted and add the `-S` argument to all `git commit` commands.
#### Motivation

It would be really annoying to not get a PR merged because of an unsigned commit.
#### Related issues

Fix: #13904
#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ A good place to learn about general guidelines for contributing to
 [Guidelines document](https://developer.mozilla.org/en-US/docs/MDN/Guidelines).
 For example, you can find out more about MDN's writing-style guidelines via the
 [Writing style guide](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide).
+Note that you will need to sign your commits with a GPG key when pushing from a local machiene and have a [GPG key](https://docs.github.com/en/authentication/managing-commit-signature-verification/adding-a-new-gpg-key-to-your-github-account) provided in settings.
```
Comment (OnkarRuikar) [inline README.md:32]: What if a user wishes to [edit a file using GitHub UI in browser](https://github.com/mdn/content/blob/main/README.md), using the pencil icon on top right, to correct a typo? And they haven't setup git and GPG on local machine. Will the user's commits have the verified badge?

If they need to have GPG key provided in GitHub settings for editing in GitHub UI also, then we need to mention that case too.

Comment (MrBrain295) [reply README.md:32]: They will.

Comment (MrBrain295) [reply README.md:32]: It will say "This commit was created on GitHub.com and signed with GitHub’s verified signature. "

**Comment by cw118** [comment]
For what it's worth, signed commits are not a requirement on the `content` repo (though I'm not sure if it'll become one).

File: README.md (hunk 1 above)
Comment (cw118) [reply README.md]: I'm a bit late here but in my experience, you don't need to have a GPG key configured when using the GitHub UI — the message "This commit was created on GitHub.com and signed with GitHub’s verified signature." confirms that the commit signature is from GitHub and not the user's.

**Comment by MrBrain295** [comment]
It says they are in #13904.

**Comment by cw118** [comment]
> It says they are in #13904.

In that case, I believe the issue is "incorrect". Since I've already pointed this (signed commits aren't required) out here, we can wait for a reviewer to verify :)

**Comment by hamishwillee** [comment]
Closing. @cw118 is correct - signing of commits is not a requirement on MDN. If it became a requirement, then yes, adding notes on how to do this would be essential.

Thanks @MrBrain295 for taking the time to take the issue on -  sorry our initial triage was wrong.

**Comment by OnkarRuikar** [comment]
@hamishwillee 
> ...  signing of commits is not a requirement on MDN. ... 

I don't believe this is true for all MDN repositories. The repo [mdn/yari](https://github.com/mdn/yari/blob/main/README.md) does require signing:
> Lastly, we require that all commits are signed. Please see the documentation [about signed commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) and [how to sign yours](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) on GitHub. [ref](https://github.com/mdn/yari/blob/main/README.md)

You have done signed commits in yari. The reason for you not needing `-S` could be that you might have enabled auto-signing:
> To configure your Git client to sign commits by default for a local repository, in Git versions 2.0.0 and above, run git config commit.gpgsign true. To sign all commits by default in any local repository on your computer, run git config --global commit.gpgsign true.  [ref](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)


I haven't checked with all other ~110 repos. I thought this was organization wide requirement. If they've done it for `mdin/yari` then they must have done it for some other repos too.

**Comment by hamishwillee** [comment]
@OnkarRuikar Thanks for making the effort to save other MDN users time. 

Since this isn't required to use `mdn/content`, and the vast majority of users do not need to touch `yari` I'm happy with my decision to close this. Happy to reopen we find that users are running into this problem a lot.

If it is a requirement for `mdn/yari` then perhaps post it against that repo?

**Comment by schalkneethling** [comment]
> If it is a requirement for `mdn/yari` then perhaps post it against that repo?

This is indeed a requirements on Yari. @MrBrain295, if we are not mentioning it in the docs over there, I would welcome a pull request to add it. Feel free to mention me for review.

**Comment by rdotterer09** [comment]
-----BEGIN PGP PUBLIC KEY BLOCK----- 

Version: BCPG C# v1.7.0.0 

 

mI0EYpOu/AEEAL9AdcL31kjx1AbaGM192ac24LGufzMUEeZ4HOGIEsaC289pGb5X 

v6hTVyv1x0BCRP5Oo5X+8SzQukHum8RIpaG7pGJUlbZpKXjcYdQOY7Rddp54+w/H 

QPVswQqK6G5slm4UjONAMjjF4Nfsy6WrF2CoZc/1DcuQ/LkwxuNETdX9ABEBAAG0 

RTI4OTk5MDM0NTgyMTc2NzUxMjg2NjM3ODk0NDI1NTYxMjA0MTE2NTYxMjAzNSA8 

cmRvdHRlcmVyMDlAZ21haWwuY29tPoicBBABAgAGBQJik678AAoJEE1nDVrwIvQr 

+eIEAJiVeBVo5ledTWEhA1CXCxqb1hECZgfrBYAVJZBZKvBCJbuwlC1hwCRuDmzq 

G8eI6mcFNXL/WdMhfqgHnIAOsLEycXyQzcnV7h6GO5kMcOHmAU9KMOHAUPgf06u8 

I5hogL0YU86t475teoS17v0Z2bwTAlBlr5DW4XTLZZ4gKVPo 

=LISa 

-----END PGP PUBLIC KEY BLOCK-----

**Comment by schalkneethling** [comment]
@rdotterer09 I deleted your comment as I am sure you did not want to post a PGP key publicly.

**Comment by rdotterer09** [comment]
> @rdotterer09 I deleted your comment as I am sure you did not want to post a PGP key publicly.

**Comment by rdotterer09** [comment]
That was the public key not the private key

**Comment by schalkneethling** [comment]
> That was the public key not the private key

Fair enough, but best not to post it on this pull request as it is not relevant.

**Pull request closed without merging**